modules/python/python-types.c

```diff
@@ -345,6 +368,31 @@ py_list_to_list(PyObject *obj, GString *list)
   return TRUE;
 }
 
+gboolean
+py_string_list_to_string_list(PyObject *obj, GList **string_list)
+{
+  if (!PyList_Check(obj))
+    {
+      PyErr_Format(PyExc_ValueError, "Error extracting value from list");
+      return FALSE;
+    }
+
+  *string_list = NULL;
+
+  for (Py_ssize_t i = 0; i < PyList_GET_SIZE(obj); i++)
+    {
+      PyObject *element = PyList_GET_ITEM(obj, i);
+
+      const gchar *element_as_str;
+      if (!py_bytes_or_string_to_string(element, &element_as_str))
+        return FALSE;
```

Comment:
we should probably clean up the partially built string_list in this case.

Reply:
Thanks, fixed!